nm/src/nm.c

```diff
@@ -264,64 +261,36 @@ _j4status_nm_format_down_other_callback(const gchar *token, guint64 value, gcons
 static void
 _j4status_nm_device_get_addresses_ipv4(J4statusPluginContext *context, NMDevice *device, GVariantBuilder *builder)
 {
-    NMIP4Config *ip4_config;
+    NMIPConfig *ip4_config;
 
     ip4_config = nm_device_get_ip4_config(device);
     if ( ip4_config == NULL )
         return;
 
-    const GSList *address_;
-    for ( address_ = nm_ip4_config_get_addresses(ip4_config) ; address_ != NULL ; address_ = g_slist_next(address_) )
+    const GPtrArray *addresses_ = nm_ip_config_get_addresses(ip4_config);
+    guint i = 0;
+    for ( i = 0 ; i < addresses_->len ; i++ )
     {
-        guint32 address;
-        address = nm_ip4_address_get_address(address_->data);
-        g_variant_builder_add_value(builder, g_variant_new_printf("%u.%u.%u.%u", (address >> 0) & 255, (address >> 8) & 255, (address >> 16) & 255, (address >> 24) & 255));
+	NMIPAddress *addr = g_ptr_array_index (addresses_, i);
+        g_variant_builder_add_value(builder, g_variant_new_string(nm_ip_address_get_address(addr)));
     }
 }
 
 static void
 _j4status_nm_device_get_addresses_ipv6(J4statusPluginContext *context, NMDevice *device, GVariantBuilder *builder)
 {
-    NMIP6Config *ip6_config;
+    NMIPConfig *ip6_config;
 
     ip6_config = nm_device_get_ip6_config(device);
     if ( ip6_config == NULL )
         return;
 
-    const GSList *address_;
-    for ( address_ = nm_ip6_config_get_addresses(ip6_config) ; address_ != NULL ; address_ = g_slist_next(address_) )
+    const GPtrArray *addresses_ = nm_ip_config_get_addresses(ip6_config);
+    guint i = 0;
+    for ( i = 0 ; i < addresses_->len ; i++ )
     {
-        const struct in6_addr *address;
-        address = nm_ip6_address_get_address(address_->data);
-
-        guint b = 0;
-        gsize o = 0;
-        gchar address_str[44]; /* ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff/128 + \0 */
-        gboolean shortened = FALSE;
-        gboolean was_shortened = FALSE;
-        guint fragment;
-        for (;;)
-        {
-            fragment = ( address->s6_addr[b] << 8 ) + address->s6_addr[b+1];
-            if ( ( fragment == 0 ) && ( ! was_shortened ) )
-                shortened = TRUE;
-            else
-            {
-                if ( shortened )
-                {
-                    o += g_snprintf(address_str, sizeof(address_str) - o, ":");
-                    shortened = FALSE;
-                    was_shortened = TRUE;
-                }
-                o += g_snprintf(address_str, sizeof(address_str) - o, "%x", fragment);
-            }
-            b += 2;
-            if ( b >= 16 )
-                break;
-            if ( ( ! shortened ) || ( b == 0 ) )
-                o += g_snprintf(address_str, sizeof(address_str) - o, ":");
-        }
-        g_variant_builder_add_value(builder, g_variant_new_string(address_str));
+	NMIPAddress *addr = g_ptr_array_index (addresses_, i);
+        g_variant_builder_add_value(builder, g_variant_new_string(nm_ip_address_get_address(addr)));
```

Comment:
Did you check that the address is properly shortened? I would guess yes, as it uses `inet_ntop` that says “in the most appropriate IPv6 network address format for this address”, but doesn’t hurt to check.

Reply:
Yes, I checked and I saw "::" in the shown addresses.

I'll send the rest later today.